pubsub/pubsub_test.go

```diff
@@ -421,3 +422,56 @@ func TestOpenCensus(t *testing.T) {
 		t.Error(diff)
 	}
 }
+
+func TestShutdownsDoNotLeakGoroutines(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	ng0 := runtime.NumGoroutine()
+	top := mempubsub.NewTopic()
+	sub := mempubsub.NewSubscription(top, time.Second)
+
+	// Send a bunch of messages at roughly the same time to make the batcher's work more difficult.
+	var wg sync.WaitGroup
+	n := 1000
+	for i := 0; i < n; i++ {
+		wg.Add(1)
+		go func() {
+			err := top.Send(ctx, &pubsub.Message{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	// Receive the messages.
+	var wg2 sync.WaitGroup
+	for i := 0; i < n; i++ {
+		wg2.Add(1)
+		go func() {
+			m, err := sub.Receive(ctx)
+			if err != nil {
+				t.Fatal(err)
```

Comment:
t.Fatal doesn't work in goroutines. Use t.Error, and find a way to exit the test from the main goroutine. Or maybe this could be an errgroup, and call t.Fatal from the main goroutine on error.

Reply:
Changed to use errgroup.

Reply:
PTAL.